### ISSUE TITLE
Problem with height and overflow

### ISSUE THREAD

**Issue by guiihlopes** [nkbt/react-collapse#182]
Hello! First of all congratulations for all this amazing work.

I'm facing a problem which appears when something inside of my nested collapse updates. It makes the height fixed of parent collapse component and makes the `overflow: hidden,` when it occurs, break my layout. Am I doing something wrong? I need the parent collapse stay with `height: auto` and don't make the `overflow: hidden`. How can I achieve this result?

Thanks!

**Comment by nkbt** [comment]
It is quite impossible not to use `overflow:hidden` when you animate hight, so you need to find a way to update your css so it does not break.

As a hint - `overflow:hidden` at some point was widely used as part of `clearfix` solution (you can get the details at https://www.quirksmode.org/css/clearing.html for example). I don't remember details anymore but it changed the way browsers render elements. So most likely in your case there is something to do with floats, and when Collapse works - it "breaks" your css. If this is true, then fix somehow your floats or switch to `flexbox` (it is supported pretty much everywhere now).

**Comment by guiihlopes** [comment]
Sorry for don't being specific enough. It's not related the CSS I guess... I have made a screen record to show both issues. The first one, I just dispatch an action of my reducer to change the text, and after it changes, just make the height of the nested collapse fixed and insert the overflow to hidden which breaks my layout.

![problemoverflow](https://user-images.githubusercontent.com/14032146/28700861-b9052038-7328-11e7-9f09-8c0ceeee73e2.gif)

And this other one screen record, is an example of my form that after I saving and dispatching an action of my reducer it turn fixes the height to the height of the actual content (because a text of something inside the nested collapse has changed) and make it overflow hidden. After that dispatch I toggle the style of my form to none and my layout is broken because the height was fixed. 

![problemwithheightoverflow](https://user-images.githubusercontent.com/14032146/28700862-b908a294-7328-11e7-8280-81f0c643908a.gif)

What could I do in this case? :/

**Comment by nkbt** [comment]
Ah, ok... So it does not break css, but cuts content. Well... You can't really do anything about it, unless you put your "popover" outside of collapsed area. Or change the design in a way that it would not be outside of collapsible area.


Though... While I am writing it I've realised that Collapse can set `overflow-y:hidden` and keep `overflow-x:visible`, which in your case should work fine. Sounds like a good idea

**Comment by guiihlopes** [comment]
Thanks for those fast answers! I'm trying to use your tips, but without success...
Setting the overflow's sounds a good idea, and it sounds to fix the first issue, but the second would still appear (look at the white space between the information and the next one collapse after it fixes the height). I don't know why it fixes my height if I'm not using `fixedHeight` prop.

Code of my nested collapse:

``` 
<Collapse isOpened hasNestedCollapse>
         <Collapse
              isOpened={this.state.isOpened}
              theme={{ content: 'accordion__content' }}
          >
          {children}
        </Collapse>
</Collapse>
```

How is the right way to set `overflow-y: hidden` and keep `overflow-x: visible`?
I have already tried: 
```
<Collapse
          isOpened={this.state.isOpened}
          **style={{
            overflowY: 'hidden',
            overflowX: 'visible',
          }}**
          theme={{ content: 'accordion__content' }}
        >
          {children}
</Collapse>
```
In this way, it don't cut if I set: 
```
style={{
   overflow: 'visible',
}}
```

But when animate it overrides the `overflow-y` to be `visible` too, and then it doesn't cut collapse content.

Also I have already tried setting in my css without success. When I set `overflow-y: hidden`, it cuts content and if I set `overflow-x: visible` nothing changes.

I don't know what to do anymore, I would love if there are any solution for those problems.

Thanks for the support until now.

**Comment by nkbt** [comment]
it has to be done in react collapse code itself, you cannot do it with your code. Can try edit source and see if it works out for you.

The second problem is most likely duplicate of #181

**Comment by nkbt** [comment]
Published `react-collapse@4.0.3` check if it solves your problem at least partially

**Comment by nkbt** [comment]
I tried using `overflow-y: hidden` and `overflow-x: visible` in all different combinations, including debugging parent elements and making sure all of them have `overflow:visible`. Unfortunately during animation, content is cut off. Must be something with the way browser's renderer works, so unfortunately I don't think it is possible to fix your case.

The only workable solution for me would be to keep popover note outside of collapsible element, so it is not cut off, and show/hide it accordingly.

**Comment by nkbt** [comment]
There are bunch of questions about it on SO like

- https://stackoverflow.com/questions/11520728/overflow-yvisible-not-working-when-overflow-xhidden-is-present

- https://stackoverflow.com/questions/6421966/css-overflow-x-visible-and-overflow-y-hidden-causing-scrollbar-issue

**Comment by guiihlopes** [comment]
Yeah, you got it! It seems to fix both problems with `react-collapse@4.0.3`.

Thanks for the support! 😄
